pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/elastic/ElasticSearchService.java

```diff
@@ -0,0 +1,103 @@
+
+/**
+ * Service for interacting with ElasticSearch
+ * </p>
+ * Note: This is based on org.broadinstitute.dsm.model.elastic.search.ElasticSearch, and we move methods from there
+ * as they get untangled from ElasticSearchUtil and come under automated testing
+ */
+@Slf4j
+public class ElasticSearchService {
+    private final ElasticSearch elasticSearch;
+
+    public ElasticSearchService() {
+        this.elasticSearch = new ElasticSearch();
+    }
+
+    /**
+     * Return a list of all participant guids in a participant ES index
+     */
+    public List<String> getAllParticipantGuids(String esParticipantsIndex) {
+        ElasticSearch es = elasticSearch.getAllParticipantsDataByInstanceIndex(esParticipantsIndex);
+        List<String> participantIds = new ArrayList<>();
+        for (ElasticSearchParticipantDto participantDto : es.getEsParticipants()) {
+            if (participantDto.getProfile().isPresent()) {
+                Profile profile = participantDto.getProfile().get();
+                if (StringUtils.isNotBlank(profile.getGuid())) {
+                    participantIds.add(profile.getGuid());
+                }
+            }
+        }
+        log.info("Got {} participant ids from index (getAllParticipantGuids)", participantIds.size());
+        return participantIds;
+    }
+
+    /**
+     * Return a list of all participant guids in a participant ES index
+     */
+    public Map<String, String> getLegacyPidsByGuid(String esIndex) {
+        SearchRequest searchRequest = new SearchRequest(esIndex);
+        SearchResponse response;
+        try {
+            SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder();
+            searchSourceBuilder.query(QueryBuilders.existsQuery(ElasticSearchUtil.PROFILE_LEGACYALTPID));
+            searchRequest.source(searchSourceBuilder);
+            response = ElasticSearchUtil.getClientInstance().search(searchRequest, RequestOptions.DEFAULT);
+        } catch (Exception e) {
+            // this is likely due to a bad query, so log it
+            log.error("Error getting ES participant legacy PIDs: index %s, query: %s"
+                    .formatted(esIndex, searchRequest));
+            throw new DsmInternalError("Error getting ES participants for instance " + esIndex, e);
+        }
+        List<ElasticSearchParticipantDto> participants = elasticSearch.parseSourceMaps(response.getHits().getHits());
+        log.info("Got {} participants from ES index {} (getLegacyPidsByGuid)", participants.size(), esIndex);
+
+        Map<String, String> guidToPid = new HashMap<>();
+        participants.forEach(participant -> {
+            if (participant.getProfile().isPresent()) {
+                Profile profile = participant.getProfile().get();
+                if (StringUtils.isNotBlank(profile.getGuid()) && StringUtils.isNotBlank(profile.getLegacyAltPid())) {
+                    guidToPid.put(profile.getGuid(), profile.getLegacyAltPid());
```

Comment:
Is it possible for us to have multiple legacy pids for the same guid?  Should we be checking that here?